ynput/ayon-nuke
# Display and View fallback options for settings and api

After adding support for context-specific anatomy token resolution in color management display and view options, we must now ensure value validation and provide fallback options. We're introducing a semicolon separator in values, allowing for multiple options where the order is important.


## Additional informations
- colorspace methods were moved from lib to new nuke API module `colorspace.py`. Some of those function might need to move into core addon but for now lets keep them here. 

## Testing notes
### Pre testing 
- version up dev number in package.py
- deploy addon version to your server and bundle
- make sure core colormanagement is enabled
- notice hints on `ayon+settings://nuke/imageio/viewer/display` 

### How to do DCC testing
- When conducting this testing, it is important to note that a published ocio config may not be present in your task context. Our objective is to determine the backward compatibility of the process and identify any potential issues.  It is possible for only certain shot folders to have published ocio configurations, while others may not. In such scenarios, a fallback option must be available. 
- In addition to this PR, the changes in https://github.com/ynput/ayon-core/pull/834 also need to be deployed. Once completed, the colorspace settings in the core addon should be adjusted to use the fallback option. The built-in **ACES 1.2** should be selected for this purpose.


1. To test baking targeting it would be good to disable clearing nodes in your code here **ayon-nuke\client\ayon_nuke\api\plugin.py:1090** - just add `#` on line start. Then after open Nuke. This way baking nodes created during **ExtractReviewIntermediates** processing will stay in script and you would be able to check how they are set. 

2. Follow next setps where we fork into two different testing
	a. _Baking Target Colorspace (Use Display & View)_
			- set `{project[code]}_{parent}_{folder[name]};sRGB` into View here **ayon+settings://nuke/imageio/baking_target/display_view/view**
			- Go to Nuke and publish render node
			- Check what is set into node just before write node.
			- it should be set to **sRGB** if ocio config is set to different then published product.
			
	b. _Baking Target Colorspace (Use Colorspace)_
			- set `{project[code]}_{parent}_{folder[name]};Output - Rec.709` into View here **ayon+settings://nuke/imageio/baking_target/colorspace**
			- Go to Nuke and publish render node
			- Check what is set into node just before write node.
			- it should be set to **Output - Rec.709** if ocio config is set to different then published product.
